doc/remailers.rst

```diff
@@ -0,0 +1,125 @@
+remailing logic:
+
+1) For all incoming messages, keep track of Autocrypt-headers. This mechanism is
+   the same that is used by clients, and requires no modifications.
+2) Inject an Autocrypt-header into outgoing messages. This header serves the
+   same purpose as usual, but adds some details for the mailing list use case.
+   The header might only be injected under certain conditions, e.g. if
+   encryption keys are available for a certain number of subscribers.
+3) When receiving an encrypted message, decrypt it and re-encrypt for individual
+   delivery to each Autocrypt-enabled subscriber. Subscribers for whom no keys
+   are available do not receive the message, but optionally a notification of
+   this circumstance (see below).
+
+The presented mechanism works well for relatively small groups of mostly active
+and Autocrypt-enabled subscribers, but becomes increasingly ineffective with
+larger numbers of passive or non Autocrypt-enabled subscribers. As with any
+encrypted group communication, the benefits of end-to-end encryption also dilute
+with larger numbers of recipients.
+
+For compatibility with clients that are not Autocrypt-capable, the remailer
+software might want to offer manual configuration of keys, e.g. via a web
+interface.
+
+Injecting Reamiler Autocrypt-Headers
+~~~~~
+
+To inform subscribers of its key, the remailer injects an Autocrypt-header into
+outgoing messages. The header format is the same as usual, where the ``addr``
+attribute is the address of the list, and ``keydata`` is their Autocrypt key.
+
+In addition to those two attributes, the remailer SHOULD add the
+``_list-recommendation`` attribute, and MAY add ``_list-recipients`` attribute.
+Both of these attributes are exclusive for use by remailers, and MUST NOT be
+used by regular mailbox addresses. Both of these attributes are non-critical
+(starting with an underscore), which will cause an Autocrypt-capable client that
+does not handle these attributes to skip over them, falling back to the usual
+Autocrypt behavior.
+
+For example::
+
+    Autocrypt: addr=post@list.example; _list-recommendation=mandatory; keydata=XXX
+    Autocrypt: addr=post@list.example; _list-recommendation=available; _list-recipients=7/15; keydata=XXX
+
+The ``_list-recommendation`` attribute SHOULD be included in an Autocrypt header
+sent by a remailer. Its value may be either ``available`` or ``mandatory``. If
+the value is set to ``mandatory``, the remailer requires encryption of every
+message, and will reject plaintext messages. If the value is ``available``,
+encryption is offered as an opt-in mechanism and plaintext messages will be
+handled as usual. In either case, encrypted messages will be delivered
+exclusively to subscribers for whom an encryption key is available.
+
+To inform the decision of whether to encrypt or not, the remailer MAY include a
+``_list-recipients`` attribute. This attribute contains the current number of
```

Comment:
There is a tradeoff in granularity of specification here. Would it be helpful to have consistency in similar terms to the `discourage` state? I feel like just leaving it up to the implementation what to make of this info is not giving enough guidance